### PR TITLE
Make sure module is properly reset before attempt to install

### PR DESCRIPTION
**Pull request by hhorak**
This reqiures https://github.com/sclorg/s2i-base-container/pull/206 be merged first.

### CODE REVIEW

**Comment by hhorak** [comment]
[test]

**Comment by hhorak** [comment]
> This reqiures [sclorg/s2i-base-container#206](https://github.com/sclorg/s2i-base-container/pull/206) be merged first.

Maybe this PR does not actually need to wait for the above (which is blocked by CI issues now), because once a change is merged in s2i-base repo, all depended community containers should get rebuild.. so if we can depend on that, merging this even now should be possible.. Still, letting @pkubatrh or @phracek to speak because I might misinterpret how the automatic community builds work.

**Comment by hhorak** [comment]
[test]

File: 10/Dockerfile.rhel8
Comment (phracek) [inline 10/Dockerfile.rhel8:52]: The code itself looks good. Tests are passing. But I don't follow the code.

Can you please describe it?
1) install and by this pull request reset nodejs module. What the reset does mean? Any docu?
2) remove nodejs package https://github.com/sclorg/s2i-nodejs-container/blob/master/10/Dockerfile.rhel8#L55
3) install nodejs package https://github.com/sclorg/s2i-nodejs-container/blob/master/10/Dockerfile.rhel8#L56

Would it be possible to add `reset` option also to https://github.com/sclorg/s2i-nodejs-container/blob/master/14/Dockerfile.rhel8#L52 version `14`?

**Review by phracek — CHANGES_REQUESTED** [review]

File: 10/Dockerfile.rhel8
Comment (hhorak) [reply 10/Dockerfile.rhel8:52]: Reset is necessary for switching module streams. So, in case there is already some nodejs module installed (thus enabled), we must first call `dnf module reset` and only then enable it. In case of version 14, the Nodejs Dockefile is built on top of s2i-core, not s2i-base, so we're sure there is no nodejs installed in the parent image and don't need to run `dnf module reset` in that case.

Comment (hhorak) [reply 10/Dockerfile.rhel8:52]: I've put the comment to the commit message as well.

**Comment by phracek** [comment]
CI is being stucked. Let's try one more.

[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Pull request merged**
